### ISSUE TITLE
Error while building Module not found: Error: Can't resolve '@material-ui/core' in '/opt/atlassian/pipelines/agent/build/node_modules/mui-validate'

### ISSUE THREAD

**Issue by aman-rawat** [kbrueckner/mui-validate#15]
> Error while building Module not found: Error: Can't resolve '@material-ui/core' in '/opt/atlassian/pipelines/agent/build/node_modules/mui-validate'

I am using @mui/material  v5.9.2

![image](https://user-images.githubusercontent.com/6269995/184303774-61935dce-53d8-402c-9047-82eb80ed15de.png)

**Comment by kbrueckner** [comment]
The problem goes back to building the bundle in a pipeline. Most of the pipeline tools let builds fail when there are warnings. I encountered similar things within gitlab ci pipelines. When building the package locally there are no issues and the final artefact is fully functional.

The solution for me  in gitlab ci was to set an environment variable
```
CI=false
```
before starting the build.

**Why is that warning/error even shown?**

To keep the package compatible with material ui v4 and v5 it tries to identify/load some components of those packages. The one version which is not available will cause that warning to be thrown. Support for material ui v4 will soon be dropped - then we will not encounter the reported behavior anymore in pipelines.

Sorry for the inconvenience in the meantime.

**Comment by kbrueckner** [comment]
Accidentally closed therefore reopening

**Comment by kbrueckner** [comment]
I have released an alpha version of version 1 which has the same feature set as 0.3.3 which is designed purely for @mui/material. It doesn't hold that logic with dynamic library loading.

Can you please check out https://www.npmjs.com/package/mui-validate/v/1.0.0-alpha.1 and see if it solves your issues and give me feedback?

**Comment by kbrueckner** [comment]
Should be fixed with version 1
